src/chromium/crBrowser.ts

```diff
@@ -50,24 +50,35 @@ export class CRBrowser extends BrowserBase {
     const connection = new CRConnection(SlowMoTransport.wrap(transport, slowMo));
     const browser = new CRBrowser(connection);
     const session = connection.rootSession;
-    const promises = [
-      session.send('Target.setDiscoverTargets', { discover: true }),
-      session.send('Target.setAutoAttach', { autoAttach: true, waitForDebuggerOnStart: true, flatten: true }),
-      session.send('Target.setDiscoverTargets', { discover: false }),
-    ];
+    if (!isPersistent) {
+      await session.send('Target.setAutoAttach', { autoAttach: true, waitForDebuggerOnStart: true, flatten: true });
+      return browser;
+    }
+
     const existingPageAttachPromises: Promise<any>[] = [];
-    if (isPersistent) {
-      // First page and background pages in the persistent context are created automatically
-      // and may be initialized before we enable auto-attach.
-      function attachToExistingPage({targetInfo}: Protocol.Target.targetCreatedPayload) {
-        if (targetInfo.type !== 'page' && targetInfo.type !== 'background_page')
-          return;
-        existingPageAttachPromises.push(session.send('Target.attachToTarget', {targetId: targetInfo.targetId, flatten: true}));
-      }
-      session.on('Target.targetCreated', attachToExistingPage);
-      Promise.all(promises).then(() => session.off('Target.targetCreated', attachToExistingPage)).catch(debugError);
+    // First page and background pages in the persistent context are created automatically
+    // and may be initialized before we enable auto-attach.
+    function attachToExistingPage({targetInfo}: Protocol.Target.targetCreatedPayload) {
+      if (targetInfo.type !== 'page' && targetInfo.type !== 'background_page')
```

Comment:
Unrelated to thus PR but this should also include service workers I believe.

Reply:
Good point, I'll fix it right away.